src/main/java/io/reactivex/Maybe.java

```diff
@@ -444,6 +857,25 @@
                 Functions.EMPTY_ACTION     // dispose
         ));
     }
+    
+    /**
+     * Calls the given onEvent callback with the (success value, null) for an onSuccess, (null, throwabe) for
+     * an onError or (null, null) for an onComplete signal from this Maybe before delivering said
+     * signal to the downstream.
+     * <p>
+     * Exceptions thrown from the callback will override the event so the downstream receives the
+     * error instead of the original signal.
+     * <dl>
+     * <dt><b>Scheduler:</b></dt>
+     * <dd>{@code doOnEvent} does not operate by default on a particular {@link Scheduler}.</dd>
+     * </dl>
+     * @param onEvent the callback to call with the terminal event tuple
+     * @return the new Maybe instance
+     */
+    public final Maybe<T> doOnEvent(BiConsumer<? super T, ? super Throwable> onEvent) {
```

Comment:
It feels weird to use `null` as such a strong signal where it's forbidden in most (all?) of the API. Not only do I prefer the readability that `Notification`'s methods provide, but it would allow the re-use of side-effecting operations (such as logging) across all stream types since they all would have the same signature.